### ISSUE TITLE
"Edit location" is hard to use (and confusing)

### ISSUE THREAD

**Issue by mnalis** [commons-app/apps-android-commons#5713]
### Feedback

When geotagged picture is being uploaded, and _"Edit location"_ button is clicked, there are several usability problems, see this video:

https://github.com/commons-app/apps-android-commons/assets/156656/325f4a12-bb14-4cf4-8631-a1fa40de6412

- `00:00:02` on opening; the map is positioned on user's current position, and not to the location of GPS EXIF Tags in the picture which would make sense (that one is already being tracked in https://github.com/commons-app/apps-android-commons/issues/5704)
- `00:00:10` even when user manually navigates to picture location and presses and presses `Show in map app`, the location selected in the other app (OsmAnd in this case) is not the location of the GPS EXIF tags from the picture, but currently viewed map center (which is going to be off, even if user tried to carefully position it in the middle). It should instead always position on exact location contained in the GPS EXIF tags.
- `00:00:34` when users presses `Edit location`, the map jump from EXIF GPS location to current center of the map, thus losing the original picture position. It should instead start from the EXIF GPS position and only move at explicit user click to move it somewhere else; as it is likely that the user only wants to move it slightly from location recorded in EXIF GPS tags (as the location of the subject of the picture is usually not the exactly same as the location of the phone, as you usually take picture of object at several (or several dozen) meters away.

### Wiki username

Mnalis

### Device name

Samsung Galaxy S23+

### Android version

Android 14 (OneUI 6.1)

### Commons app version

5.0.1~af028cbdd (latest F-droid)

**Comment by Jason-Whitmore** [comment]
Can I be assigned this issue? I already have a pull request which hopefully resolves issue #5704 and I am familiar with the code that is probably causing the rest of the issues.

**Comment by Jason-Whitmore** [comment]
@mnalis I'm having trouble reproducing the third problem. When I follow the steps you provided, the map will center near the EXIF location, not the center of the map. I believe this is also happening in the video you uploaded.

I recommend trying to move the map a large distance (a kilometer) away from the pin before tapping the blue "Edit Location" button. Does the problem still occur?

Thank you.

**Comment by mnalis** [comment]
> I recommend trying to move the map a large distance (a kilometer) away from the pin before tapping the blue "Edit Location" button. Does the problem still occur?

You are correct, when current GPS location is several kilometers away from EXIF location, clicking "Edit Location" does seem to move the map towards (or at least very near) EXIF location. I'm eagerly awaiting next release to test out your fixes @Jason-Whitmore, so I'll do more check then to confirm if everything seems fine!
